api/src/database/channel-data.ts

```diff
@@ -0,0 +1,38 @@
+import { CollectionNames } from './constants';
+import { MongoDbService } from '../services/mongodb-service';
+import { ChannelData } from '../models/types/channel-data';
+
+const collectionName = CollectionNames.channelData;
+const getIndex = (link: string, userId: string) => `${link}-${userId}`;
+
+export const getChannelData = async (channelAddress: string, userId: string, limit?: number, index?: number): Promise<any> => {
+	const query = { channelAddress, userId };
+	const skip = index > 0 ? (index - 1) * limit : 0;
```

Comment:
I didn't get why do you multiply by the limit value

Reply:
because if the limit is for instance 100 and index is 2 it shall skip 100 so it is able to get all data from index 101

Reply:
I will change the index to start at 0, this is something I have thought of because the behaviour was a bit confusing

Reply:
shouldn't it be `(index - 1) + limit` in this case? 
because otherwise for limit 100 and index 5 you will skip 400 messages

Reply:
I've changed it to: index * limit but in general:

for index 5 and limit 100 I would have expected to skip 500 messages yes. In fact this is a pagination option where each page can show 100 list entries, so for the 5th index you want to skip all the prev 500 messages.

index + limit would lead to a skip of: 105 messages